app/models/dish.rb

```diff
@@ -16,8 +16,41 @@ class Dish < ActiveRecord::Base
   RATE_DIMENSION_DISH = 'dish'
   ratyrate_rateable RATE_DIMENSION_DISH
 
+  def rates
+    Rate.where(rateable_id: id, dimension: RATE_DIMENSION_DISH)
+  end
+
   def raters_count
-    Rate.where(rateable_id: id, dimension: RATE_DIMENSION_DISH).pluck(:rater_id).uniq.count
+    rates.pluck(:rater_id).uniq.count
+  end
+
+  def rating
+    rates.average(:stars) || -1
+  end
+
+  def search_data
+    if user && user.location && user.location.is_visible
```

Comment:
@xudongcamsys Sorry I got that a bit wrong. I wrote that from memory without trying it out in the console, which is what I should've done in the first place.

The method name is `readonly!` with a bang, here are the docs: http://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-readonly-21

However `readonly!` returns a boolean, so the code I wrote still wouldn't be quite right, as `DEFAULT_LOCATION` would end up being a boolean not a `Location` object.

I tried a few things out in the rails console and I recommend you also `freeze` the object to prevent the `Location` attributes from being modified after initialization:

```ruby
  ANONYMOUS_LOCATION = Location.new(lat: 1000, lng: 1000).freeze.tap(&:readonly!)
```

*( in case you haven't seen the `tap` API docs: http://ruby-doc.org/core-2.2.2/Object.html#method-i-tap )*